sources/osgStats/Stats.js

```diff
@@ -0,0 +1,299 @@
+var MACROUTILS = require('osg/Utils');
+var notify = require('osg/notify');
+var BufferArray = require('osg/BufferArray');
+var BlendFunc = require('osg/BlendFunc');
+var Camera = require('osg/Camera');
+var CullFace = require('osg/CullFace');
+var Depth = require('osg/Depth');
+var Node = require('osg/Node');
+var Program = require('osg/Program');
+var Shader = require('osg/Shader');
+var Transform = require('osg/Transform');
+var Texture = require('osg/Texture');
+var mat4 = require('osg/glMatrix').mat4;
+var Buffer = require('osgStats/Buffer');
+var Counter = require('osgStats/Counter');
+var TextGenerator = require('osgStats/TextGenerator');
+
+var Stats = function(viewport) {
+    this._captionsBuffer = undefined;
+    this._valuesBuffer = undefined;
+
+    this._dirtyCaptions = true;
+    this._dirtyValues = true;
+
+    this._counters = undefined;
+    this._groups = undefined;
+
+    this._labelMaxWidth = 0;
+    this._viewport = viewport;
+
+    this._node = undefined;
+    this._text = undefined;
+    this._offsetGroup = undefined;
+
+    this._counters = {};
+    this._groups = [];
+    this._updates = [];
+
+    this._x = 0;
+    this._y = 0;
+    this._width = 0;
+    this._height = 0;
+
+    this._fontFactor = 0.7;
+    this._lineFactor = 1.1;
+    this._characterDisplayHeight = 32.0;
+
+    this._displayFilter = [];
+    this._init(viewport);
+};
+
+MACROUTILS.createPrototypeObject(Stats, {
+    getCounter: function(name) {
+        return this._counters[name];
+    },
+    addConfig: function(config) {
+        for (var valueName in config.values) {
+            if (this._counters[valueName]) {
+                notify.error('Counter ' + valueName + ' already exist');
+                return;
+            }
+            var valueConfig = config.values[valueName];
+            var counter = new Counter(valueConfig);
+            this._counters[valueName] = counter;
+        }
+
+        if (config.groups) {
+            for (var i = 0; i < config.groups.length; i++) {
+                this._groups.push(config.groups[i]);
+            }
+        }
+
+        if (config.update) {
+            this._updates.push(config.update);
+        }
+    },
+    update: function() {
+        for (var i = 0; i < this._updates.length; i++) {
+            this._updates[i](this);
+        }
+
+        this._dirtyValues = true;
```

Comment:
I wonder if we could throttle that to One update per Second to limit the update perf cost while still getting nice stats.